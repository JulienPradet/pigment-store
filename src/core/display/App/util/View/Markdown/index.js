import React from 'react'
import Prism from 'prismjs'
import 'prismjs/components/prism-jsx'
import markdown from 'markdown-it'

const md = markdown({
  html: false,
  xhtmlOut: false,
  breaks: false,
  linkify: false,
  typographer: false,
  quotes: '“”‘’',
  highlight: function (code, lang = 'jsx') {
    if (!Prism.languages.hasOwnProperty(lang)) {
      lang = 'jsx'
    }

    return `<pre class="language-${lang}"><code>${Prism.highlight(code, Prism.languages[lang])}</code></pre>`
  }
})

export default class Markdown extends React.Component {
  constructor (props) {
    super()
    this.state = {
      content: props.children
        ? md.render(typeof props.children === 'string' ? props.children : props.children.join('\n'))
        : ''
    }
  }

  render () {
    return <div dangerouslySetInnerHTML={{ __html: this.state.content }} />
  }
}
