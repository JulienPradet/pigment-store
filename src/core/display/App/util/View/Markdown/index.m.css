.container pre {
  margin: 1em 0;
}
