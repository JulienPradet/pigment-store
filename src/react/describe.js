import describe from './definition/Component'

export default describe
