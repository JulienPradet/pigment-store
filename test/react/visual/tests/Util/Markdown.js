import React from 'react'
import describe from 'pigment-store/react/describe'
import Markdown from '../../../../../src/core/display/App/util/View/Markdown/index'

export default describe('Markdown', Markdown)
  .featureJsx(
    'Default',
    () => (
      <Markdown>
        {`# Hi!

  Here is some **basic** markdown`}
      </Markdown>
    )
  )
  .featureJsx(
    'With code',
    () => (
      <Markdown>
        ```jsx
        {`// Comment
  <div class="test">HTML</div>`}
        ```
      </Markdown>
    )
  )
