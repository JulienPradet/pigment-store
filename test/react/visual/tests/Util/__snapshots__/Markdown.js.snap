exports[`Markdown Default 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h1>Hi!</h1>
    <p>Here is some <strong>basic</strong> markdown</p>
    ",
    }
  } />
`;

exports[`Markdown With code 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<pre class=\"language-jsx\"><code><span class=\"token comment\" spellcheck=\"true\">// Comment</span>
    <span class=\"token tag\"><span class=\"token tag\"><span class=\"token punctuation\">&lt;</span>div</span> <span class=\"token attr-name\">class</span><span class=\"token attr-value\"><span class=\"token punctuation\">=</span><span class=\"token punctuation\">\"</span>test<span class=\"token punctuation\">\"</span></span><span class=\"token punctuation\">></span></span>HTML<span class=\"token tag\"><span class=\"token tag\"><span class=\"token punctuation\">&lt;/</span>div</span><span class=\"token punctuation\">></span></span>
    </code></pre>
    ",
    }
  } />
`;
